pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/MessageIdImpl.java

```diff
@@ -77,8 +77,7 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        return Objects.toStringHelper(MessageIdImpl.class).add("ledgerId", ledgerId).add("entryId", entryId)
-                .add("partitionIndex", partitionIndex).toString();
+        return String.format("%d:%d:%d", ledgerId, entryId, partitionIndex);
```

Comment:
@jai1 One last ask.. If the message is not coming from a partition, we should omit the index: 

```java
if (partitionIndex == -1) {
    return String.format("%d:%d", ledgerId, entryId);
} else {
    return String.format("%d:%d:%d", ledgerId, entryId, partitionIndex);
}
```